driver-core/src/main/com/mongodb/connection/netty/NettyStream.java

```diff
@@ -191,15 +198,7 @@ private void initializeChannel(final AsyncCompletionHandler<Void> handler, final
                 public void initChannel(final SocketChannel ch) {
                     ChannelPipeline pipeline = ch.pipeline();
                     if (sslSettings.isEnabled()) {
-                        SSLEngine engine = getSslContext().createSSLEngine(address.getHost(), address.getPort());
-                        engine.setUseClientMode(true);
-                        SSLParameters sslParameters = engine.getSSLParameters();
-                        enableSni(address.getHost(), sslParameters);
-                        if (!sslSettings.isInvalidHostNameAllowed()) {
-                            enableHostNameVerification(sslParameters);
-                        }
-                        engine.setSSLParameters(sslParameters);
-                        pipeline.addFirst("ssl", new SslHandler(engine, false));
```

Comment:
This code was moved to `addSslHandler` almost as it was. Thee only modification is with regard to how we get `SSLEngine`.